### PR TITLE
chore: Drop support for Python 3.7 on kedro-datasets

### PR DESCRIPTION
**Pull request by lrcouto**
## Description

Drops support for Python 3.7 on `kedro-datasets`. Part of https://github.com/kedro-org/kedro/issues/2158.

## Checklist

- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: kedro-datasets/setup.py [hunk 1]
```diff
@@ -155,7 +155,6 @@ def _collect_requirements(requires):
     "Jinja2<3.1.0",
 ]
 extras_require["test"] = [
-    "adlfs>=2021.7.1, <=2022.2; python_version == '3.7'",
```
Comment (astrojuanlu) [inline kedro-datasets/setup.py:158]: 👍🏽 

And `>=3.8` markers are redundant and could be dropped too.

**Review by astrojuanlu — COMMENTED** [review]
Is the plan to have one last release with 3.8+ support before applying https://github.com/kedro-org/kedro-plugins/issues/404 ? Or should we go directly to it and drop 3.8 for kedro-datasets 2.0? @merelcht

**Comment by merelcht** [comment]
> Is the plan to have one last release with 3.8+ support before applying #404 ? Or should we go directly to it and drop 3.8 for kedro-datasets 2.0? @merelcht

IMO we should also drop 3.8 now we have decided to use NEP 29 for `kedro-datasets`. We can do it in a separate PR if we want to have a better discussion there.

**Review by merelcht — APPROVED** [review]
LGTM! 👍

**Review by astrojuanlu — APPROVED** [review]
💯

**Pull request merged**